### ISSUE TITLE
freetype: packed struct can be bitcast, no need for utils

### ISSUE THREAD

**Issue by mitchellh** [hexops/mach#490]
Perhaps I am missing a reason for this, so I apologize if that's the case and I'd appreciate being enlightened. I noticed in your bitfields, you use a `utils` package with some clever comptime stuff to convert to/from bitfields. This is unnecessary, if your bitfields are in a `packed struct` (they are) and they match the size of the freetype type (they don't, but they can), you can just `@bitCast`.

Here is the `freetype.LoadFlags` struct from my bindings, with tests to show you:

```zig
/// A list of bit field constants for FT_Load_Glyph to indicate what kind of
/// operations to perform during glyph loading.
pub const LoadFlags = packed struct {
    no_scale: bool = false,
    no_hinting: bool = false,
    render: bool = false,
    no_bitmap: bool = false,
    vertical_layout: bool = false,
    force_autohint: bool = false,
    crop_bitmap: bool = false,
    pedantic: bool = false,
    ignore_global_advance_with: bool = false,
    no_recurse: bool = false,
    ignore_transform: bool = false,
    monochrome: bool = false,
    linear_design: bool = false,
    no_autohint: bool = false,
    _padding: u1 = 0,
    target_normal: bool = false,
    target_light: bool = false,
    target_mono: bool = false,
    target_lcd: bool = false,
    target_lcd_v: bool = false,
    color: bool = false,
    compute_metrics: bool = false,
    bitmap_metrics_only: bool = false,
    _padding2: u9 = 0,

    test {
        // This must always be an i32 size so we can bitcast directly.
        const testing = std.testing;
        try testing.expectEqual(@sizeOf(i32), @sizeOf(LoadFlags));
    }

    test "bitcast" {
        const testing = std.testing;
        const cval: i32 = c.FT_LOAD_RENDER | c.FT_LOAD_PEDANTIC;
        const flags = @bitCast(LoadFlags, cval);
        try testing.expect(flags.render);
        try testing.expect(flags.pedantic);
    }
};
```

This has been working fine for me for awhile, but was curious since we do it rather differently if there is a reason you don't do this?

**Comment by alichraghi** [comment]
Thanks for telling. that's really cool! i just didn't know about it (sorry)

**Comment by slimsag** [comment]
I should've mentioned this during review :) we use the same packed struct approach in the `mach/gpu` WebGPU API and in `mach/glfw` bindings

**Comment by slimsag** [comment]
Fixed by #487
